eta/core/datasets.py

```diff
@@ -1812,33 +1814,14 @@ def build(self, path, description=None, pretty_print=False,
         )
 
         dataset = self.dataset_cls.create_empty_dataset(path, description)
-        dataset_dir = os.path.dirname(path)
-        data_subdir = os.path.join(dataset_dir, dataset._DATA_SUBDIR)
-        labels_subdir = os.path.join(dataset_dir, dataset._LABELS_SUBDIR)
+        data_subdir = os.path.join(dataset.data_dir, dataset._DATA_SUBDIR)
+        labels_subdir = os.path.join(dataset.data_dir, dataset._LABELS_SUBDIR)
 
         for record in self._dataset:
             data_filename = os.path.basename(record.new_data_path)
             labels_filename = os.path.basename(record.new_labels_path)
-
-            # add an incrementing index to the filename until a unique name
-            # is found
-            data_basename, data_ext = os.path.splitext(data_filename)
-            labels_basename, labels_ext = os.path.splitext(labels_filename)
-            idx = -1
-            while True:
-                idx += 1
-                unique_appender = "-{}".format(idx)
-                data_path = os.path.join(
-                    data_subdir,
-                    data_basename + unique_appender + data_ext
-                )
-                if dataset.has_data_with_name(data_path):
-                    continue
-                labels_path = os.path.join(
-                    labels_subdir,
-                    labels_basename + unique_appender + labels_ext
-                )
-                break
```

Comment:
This is still relevant if:
- a video is split into clips
- two datasets with the same name are merged
- third thing that we haven't thought of

This isn't intended to be something is used, but it does act as a safeguard

Reply:
The problem is that it always appends an index, even if you're not merging datasets at all, for example. This makes it a bit of pain to match filenames with the ones in the original dataset.  How about having logic that only adds indices once a duplicate name is found?

Reply:
(If you agree with the approach of logic that only adds indices once a duplicate name is found, I can add that.)

Reply:
I don't know how or when this was changed. The original logic is here:
https://github.com/voxel51/eta/pull/343/files
It uses the method `has_data_with_name` in the while loop

Reply:
It looks like the while loop needs to be fixed, and the `data_path` and `labels_path` should be initialized before entering the while loop, such that the while loop simply modifies `data_path` and `labels_path` if necessary.

Reply:
These were my changes. Apologies.

Reply:
I think we're all sorted out now!